src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpLexer.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.DotNet.Interactive.HttpRequest;
+
+internal class HttpLexer
+{
+    private readonly SourceText _sourceText;
+    private readonly HttpSyntaxTree _syntaxTree;
+    private TextWindow? _textWindow;
+    private readonly List<HttpSyntaxToken> _tokens = new();
+
+    public HttpLexer(SourceText sourceText, HttpSyntaxTree syntaxTree)
+    {
+        _sourceText = sourceText;
+        _syntaxTree = syntaxTree;
+    }
+
+    public IReadOnlyList<HttpSyntaxToken> Lex()
+    {
+        _textWindow = new TextWindow(0, _sourceText.Length);
+
+        HttpTokenKind? previousTokenKind = null;
+
+        char previousCharacter = default;
+
+        while (More())
+        {
+            var currentCharacter = _sourceText[_textWindow.End];
+
+            var currentTokenKind = currentCharacter switch
+            {
+                ' ' or '\t' => HttpTokenKind.Whitespace,
+                '\n' or '\r' or '\v' => HttpTokenKind.NewLine,
+                _ => Char.IsPunctuation(currentCharacter)
+                         ? HttpTokenKind.Punctuation
+                         : HttpTokenKind.Word,
+            };
+
+            if (previousTokenKind is { } previousTokenKindValue)
+            {
+                if (currentTokenKind is HttpTokenKind.NewLine && previousTokenKindValue is HttpTokenKind.NewLine 
+                    && previousCharacter is '\r' && currentCharacter is '\n')
+                {
+
```

Comment:
It would be nice to clean this up a bit to avoid the empty if block. Consider extracting a local helper (e.g., `IsCurrentTokenANewLinePreceededByCarriageReturn)` so that the resulting combined condition is a bit easier to understand.